chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -402,40 +424,34 @@ func (b *BitcoindNotifier) handleRelevantTx(tx chain.RelevantTx, bestHeight int3
 		return
 	}
 
-	// First, check if this transaction spends an output
-	// that has an existing spend notification for it.
-	for i, txIn := range msgTx.TxIn {
+	spendingTx := tx.TxRecord.MsgTx
+	spendingTxHash := spendingTx.TxHash()
+	activeSpendNtfns := b.txNotifier.ActiveSpendNotifications()
```

Comment:
The diff is gone now, but I think it could be racy since after you get the active spend notifications, you have no guarantee no more are added to the active set while using those you fetched.